workers/unity/Packages/io.improbable.gdk.core/NetworkStats/NetStats.cs

```diff
@@ -0,0 +1,160 @@
+        private readonly Dictionary<(uint, uint), DataPoint> commandRequests = new Dictionary<(uint, uint), DataPoint>();
+        private readonly Dictionary<(uint, uint), DataPoint> commandResponses = new Dictionary<(uint, uint), DataPoint>();
+        private readonly Dictionary<WorldCommand, DataPoint> worldCommandRequests = new Dictionary<WorldCommand, DataPoint>();
+        private readonly Dictionary<WorldCommand, DataPoint> worldCommandResponses = new Dictionary<WorldCommand, DataPoint>();
+
+        private NetStats()
+        {
+        }
+
+        [Conditional("UNITY_EDITOR")]
+        public void AddUpdate(in ComponentUpdate update)
+        {
+            var componentId = update.ComponentId;
+            var size = update.SchemaData.Value.GetFields().GetWriteBufferLength() +
+                update.SchemaData.Value.GetEvents().GetWriteBufferLength();
+
+            updates.TryGetValue(componentId, out var metrics);
+            metrics.Count += 1;
+            metrics.Size += size;
+            updates[componentId] = metrics;
+        }
+
+        [Conditional("UNITY_EDITOR")]
+        public void AddCommandRequest(in CommandRequest request)
+        {
+            var componentId = request.ComponentId;
+            var commandIndex = request.CommandIndex;
+            var size = request.SchemaData.Value.GetObject().GetWriteBufferLength();
+
+            var key = (componentId, commandIndex);
+            commandRequests.TryGetValue(key, out var metrics);
+            metrics.Count += 1;
+            metrics.Size += size;
+            commandRequests[key] = metrics;
+        }
+
+        [Conditional("UNITY_EDITOR")]
+        public void AddCommandResponse(in CommandResponse response, string message)
+        {
+            var componentId = response.ComponentId;
+            var commandIndex = response.CommandIndex;
+            uint size;
+
+            if (response.SchemaData.HasValue)
+            {
+                size = response.SchemaData.Value.GetObject().GetWriteBufferLength();
+            }
+            else
+            {
+                // Approximation of on-wire size.
+                size = (uint) System.Text.Encoding.UTF8.GetByteCount(message);
+            }
+
+            var key = (componentId, commandIndex);
+            commandResponses.TryGetValue(key, out var metrics);
+            metrics.Count += 1;
+            metrics.Size += size;
+            commandResponses[key] = metrics;
+        }
+
+        [Conditional("UNITY_EDITOR")]
+        public void AddWorldCommandRequest(WorldCommand command)
+        {
+            worldCommandRequests.TryGetValue(command, out var metrics);
+            metrics.Count += 1;
+            worldCommandRequests[command] = metrics;
+        }
+
+        [Conditional("UNITY_EDITOR")]
+        public void AddWorldCommandResponse(WorldCommand command)
+        {
+            worldCommandResponses.TryGetValue(command, out var metrics);
+            metrics.Count += 1;
```

Comment:
why don't we track the size of world command requests/responses?

Reply:
We don't know what the on-wire size is, there is no API to get this 😕